client/components/site-selector/index.jsx

```diff
@@ -272,8 +267,7 @@ class SiteSelector extends Component {
 	renderSites() {
 		let sites;
 
-		// Assume that `sites` is falsy when loading
-		if ( this.props.isRequestingMissingSites ) {
+		if ( ! this.props.hasLoadedSites ) {
```

Comment:
Are we ever going to hit this case? When sites are being loaded, you can't access the Site Selector. On the other hand, if we went with the approach in #21329, the placeholder wouldn't be shown as there would already be at least a single site in `sites.items` by the time you open Site Selector.

Reply:
If we introduce the loading of recent sites first, we'll redo this condition. Let's merge for now.

Reply:
> Are we ever going to hit this case?

In the site selector in sidebar, I think we won't. If the sites list is not available yet, the `CurrentSite` component will display a "Loading My Sites..." placeholder and no "Switch Site" button. So there's no way how to even show the `SiteSelector`.

But the `SiteSelector` is used at many other places (just grep the sources to see them), where you most likely can get to see the placeholder. I didn't test everything that thoroughly though.

Reply:
> if we went with the approach in #21329, the placeholder wouldn't be shown as there would already be at least a single site

Yes, that's true. Also, many other functions assume that the sites list contains the full site list, although possibly outdated. Not just one site. For example, `hasJetpackSites` selector returns reasonable results only when operating on a full site list.